File: test/data/go/css_selectors_test.go
package geometry

import "fmt"

type Point struct {
	X float64
	Y float64
}

func NewPoint(x, y float64) Point {
	return Point{X: x, Y: y}
}

func (p Point) Distance(other Point) float64 {
	dx := p.X - other.X
	dy := p.Y - other.Y
	return dx*dx + dy*dy
}

func (p Point) String() string {
	return fmt.Sprintf("(%f, %f)", p.X, p.Y)
}

type Shape interface {
	Area() float64
	Perimeter() float64
}

type Rectangle struct {
	Width  float64
	Height float64
}

func (r Rectangle) Area() float64 {
	return r.Width * r.Height
}

func (r Rectangle) Perimeter() float64 {
	return 2 * (r.Width + r.Height)
}

func ProcessShapes(shapes []Shape) {
	for _, shape := range shapes {
		if shape.Area() > 0 {
			fmt.Println(shape.Area())
		}
	}
}

func Helper(val float64) float64 {
	if val > 0 {
		return val * 2
	}
	return 0
}

File: test/data/rust/css_selectors_test.rs
// Test fixture for CSS selector coverage

pub struct Point {
    x: f64,
    y: f64,
}

impl Point {
    pub fn new(x: f64, y: f64) -> Self {
        Point { x, y }
    }

    pub fn distance(&self, other: &Point) -> f64 {
        ((self.x - other.x).powi(2) + (self.y - other.y).powi(2)).sqrt()
    }

    fn is_origin(&self) -> bool {
        self.x == 0.0 && self.y == 0.0
    }
}

pub trait Drawable {
    fn draw(&self);
    fn visible(&self) -> bool;
}

impl Drawable for Point {
    fn draw(&self) {
        println!("Point({}, {})", self.x, self.y);
    }

    fn visible(&self) -> bool {
        true
    }
}

pub fn create_point(x: f64, y: f64) -> Point {
    Point::new(x, y)
}

fn helper(val: f64) -> f64 {
    if val > 0.0 {
        val * 2.0
    } else {
        0.0
    }
}

pub mod geometry {
    pub fn area(width: f64, height: f64) -> f64 {
        width * height
    }

    pub fn perimeter(width: f64, height: f64) -> f64 {
        2.0 * (width + height)
    }
}
